Server/src/controllers/fileUploadController.ts

```diff
@@ -1,32 +1,40 @@
-const fileUploadService = require('../services/fileUploadService');
-const fs = require('fs');
+import { Request, Response } from 'express';
+import { fileUploadService } from '../services/fileUploadService';
+
+
 /**
  * The fileUploadController is resposible for providing instructions to the application if a request comes in
  * to the /dataupload api. This controller will call the Service for appropriate processing of the input. The
  * controller is resposible for providing a response back to the Client on success. The fileUploadController 
- * creates a request for the fileUploadService to process, only passing the path of the JSON.
+ * creates a request for the fileUploadService to process, only passing the path of the file.
  */
 
-const createRequest = async (req, res) => {
-  if (!req.body) {
-    res.status(400).send({
-      message: "Nothing to process"
-    });
+export class fileUploadController {
+  private filePathOfUpload: string;
+
+  constructor(filePath: string) {
+    this.filePathOfUpload = filePath;
   }
-  else {
-    try {
-      callFileUploadService(req.file.path, res).catch;
-    } catch (error) {
-      console.error(error)
+
+  createRequest(request: Request, response: Response /**nextFunction: NextFunction */) {
+    //TODO: Custom validations here 
+    if (!request.body) {
+      response.status(400).send({
+        message: "Nothing to process"
```

Comment:
I would change the message body to "Received request with empty body"

Reply:
Handled with error ticket